merlin/dataloader/loader_base.py

```diff
@@ -569,22 +581,34 @@ def _handle_tensors(self, tensors, tensor_names):
                 tensor, lists = tensor
 
             names = [i for i in names if i not in lists]
+            if not self.lists_as_tuple:
+                names_list = list(lists.keys())
+                for col in names_list:
+                    tup = lists.pop(col)
+                    X.update({f"{col}__values": tup[0]})
+                    X.update({f"{col}__offsets": tup[1]})
 
             # now add in any scalar tensors
             if len(names) > 1:
                 tensors = self._tensor_split(tensor, len(names), axis=1)
                 # pull out label tensor from here
                 lists.update(zip(names, tensors))
             elif len(names) == 1:
-                lists[names[0]] = tensor
+                lists[names[0]] = self._reshape_dim(tensor)
             X.update(lists)
 
         for column_name in self.sparse_names:
             if column_name in self.sparse_max:
                 # raise ValueError(
                 #     f"Did not convert {column_name} to sparse due to missing sparse_max entry"
                 # )
-                X[column_name] = self._to_sparse_tensor(X[column_name], column_name)
+                if self.lists_as_tuple:
+                    tensor = (X[column_name][0], X[column_name][1][:-1])
```

Comment:
It seems like we want the full set of offsets here, not just the last one (`[:-1]`)

Reply:
Currently, the PyTorch dataloader does NOT provide the last offsets (length of a batch). See comment: https://github.com/NVIDIA-Merlin/dataloader/pull/101#discussion_r1113399758

The function `_to_sparse_tensor` (actually I think the result is a dense_tensor) does not work, when the last offset (length) is part of the offset tensors.

Reply:
Ohhhhh, I get it now 👍🏻

Reply:
I noticed the issue was that the torch dataloader adds the last offset in the _to_sparse_tensor:
https://github.com/NVIDIA-Merlin/dataloader/blob/main/merlin/dataloader/torch.py#L148-L151

I will remove these lines in my updated PR